client/src/map/shareLocationDropdown.tsx

```diff
@@ -20,6 +20,11 @@ const DropdownContent = styled.div`
    * (since links don't have rounded corners)
    */
   overflow: hidden;
+
+  /* hack for collapsed nav button positioning */
+  @media (max-width: 750px) {
+    margin-left: -65px;
```

Comment:
One point, though. I think this offset should be bigger. A suitable value seems like -99px. At least that seems better on my machine.